### PR TITLE
Blended program fixes 4.8.4

### PR DESCRIPTION
**Pull request by juhiagl8**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions: Java-11, play2-2.7.2, scala-2.11, redis-5.0.3
* Hardware versions: 2 CPU / 4GB RAM

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/responsecode/ResponseMessage.java [hunk 1]
```diff
@@ -460,6 +460,7 @@ interface Message {
     String MISSING_FIXED_BATCH_ID = "Missing Fixed Batch Id.";
     String CANNOT_UPDATE_EVENT_SET_HAVING_ENROLLMENTS = "Cannot update event set having enrollments.";
     String USER_DOES_NOT_HAVE_ACCESS = "User doesn't have access to this Course Id";
+    String CURRENT_BATCH_SIZE_PARAMETER = "currentBatchSize param missing in request. Please check the incoming request";
```
Comment (karthik-tarento) [inline course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/responsecode/ResponseMessage.java]: Request body is missing parameter 'currentBatchSize' in batchAttributes.

File: course-mw/course-actors-common/src/main/java/org/sunbird/learner/actors/certificate/service/CourseBatchCertificateActor.java [hunk 2]
```diff
@@ -148,6 +148,19 @@ private Map<String, Object> mapESFieldsToObject(Map<String, Object> courseBatch)
...
+    if(MapUtils.isNotEmpty(batchAttributes)){
+      batchAttributes
+              .entrySet()
+              .stream()
+              .forEach(
+                      batchAttribute ->
+                              batchAttributes.put(
+                                      batchAttribute.getKey(), mapToObject(batchAttribute.getValue())));
+      courseBatch.put(CourseJsonKey.BATCH_ATTRIBUTES, batchAttributes);
+    }
```
Comment (karthik-tarento) [inline course-mw/course-actors-common/src/main/java/org/sunbird/learner/actors/certificate/service/CourseBatchCertificateActor.java]: this requires changes in mapToObject method too...

File: course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 3]
```diff
@@ -652,6 +657,17 @@ private CourseBatch mapESFieldsToObject(CourseBatch courseBatch) {
...
+    if(MapUtils.isNotEmpty(batchAttributes)) {
+      batchAttributes
+              .entrySet()
+              .stream()
+              .forEach(
+                      batchAttribute ->
+                              batchAttributes.put(
+                                      batchAttribute.getKey(), mapToObject((Map<String, Object>) batchAttribute.getValue())));
+      courseBatch.setBatchAttributes(batchAttributes);
+    }
```
Comment (karthik-tarento) [inline course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java]: this requires changes in mapToObject method too.

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/responsecode/ResponseMessage.java [hunk 4]
```diff
@@ -848,5 +849,6 @@ interface Key {
     String MISSING_FIXED_BATCH_ID = "MISSING_FIXED_BATCH_ID";
     String CANNOT_UPDATE_EVENT_SET_HAVING_ENROLLMENTS = "CANNOT_UPDATE_EVENT_SET_HAVING_ENROLLMENTS";
     String USER_DOES_NOT_HAVE_ACCESS = "USER_DOES_NOT_HAVE_ACCESS";
+    String CURRENT_BATCH_SIZE_MISSING =  "CURRENT_BATCH_SIZE_MISSING";
```
Comment (karthik-tarento) [inline course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/responsecode/ResponseMessage.java]: MISSING_FIELD_CURRENT_BATCH_SIZE

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

**Review by karthik-tarento — APPROVED** [review]

File: course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 5]
```diff
@@ -116,6 +116,11 @@ private void createCourseBatch(Request actorMessage) throws Throwable {
     validateContentOrg(actorMessage.getRequestContext(), courseBatch.getCreatedFor());
     validateMentors(courseBatch, (String) actorMessage.getContext().getOrDefault(JsonKey.X_AUTH_TOKEN, ""), actorMessage.getRequestContext());
     courseBatch.setBatchId(courseBatchId);
+    String primaryCategory = (String) contentDetails.getOrDefault(JsonKey.PRIMARYCATEGORY, "");
+    if (JsonKey.PRIMARY_CATEGORY_BLENDED_PROGRAM.equalsIgnoreCase(primaryCategory) && courseBatch.getBatchAttributes().get(JsonKey.CURRENT_BATCH_SIZE) == null) {
```
Comment (wilkysingh-tarento) [inline course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java:120]: Add one more condition that currentBatchSize must be greater than 0 if present

File: course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 6]
```diff
@@ -116,6 +116,11 @@ private void createCourseBatch(Request actorMessage) throws Throwable {
...
+      ProjectCommonException.throwClientErrorException(
```
Comment (wilkysingh-tarento) [inline course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java:121]: Please check are we using this common exception already

**Pull request merged**